packages/components/src/internal/util/Date.ts

```diff
@@ -61,6 +65,22 @@ export function isDateTimeCol(col: QueryColumn): boolean {
     return false;
 }
 
+export function isTimeCol(col: QueryColumn): boolean {
```

Comment:
I suggest we only have one place where we determine if a `QueryColumn` is a time column. `QueryColumn.isTimeColumn` seems the most appropriate.

All the usages of this utility are within `DatePickerInput` so it should be a fairly straightforward refactor.